poly/src/domain/radix2/fft.rs

```diff
@@ -20,6 +20,46 @@ enum FFTOrder {
 }
 
 impl<F: FftField> Radix2EvaluationDomain<F> {
+    /// Degree aware FFT that runs in O(n log d) instead of O(n log n)
+    /// Implementation copied from libiop.
+    pub(crate) fn degree_aware_fft_in_place<T: DomainCoeff<F>>(&self, coeffs: &mut Vec<T>) {
+        if !self.offset.is_one() {
+            Self::distribute_powers(&mut *coeffs, self.offset);
+        }
+        let n = self.size();
+        let log_n = self.log_size_of_group;
+        let num_coeffs = coeffs.len();
+        let log_d = ark_std::log2(num_coeffs);
+        // When the polynomial is of size k*|coset|, for k < 2^i,
+        // the first i iterations of Cooley Tukey are easily predictable.
+        // This is because they will be combining g(w^2) + wh(w^2), but g or h will always refer
+        // to a coefficient that is 0.
+        // Therefore those first i rounds have the effect of copying the evaluations into more locations,
+        // so we handle this in initialization, and reduce the number of loops that are performing arithmetic.
+        // The number of times we copy each initial non-zero element is as below:
+        let duplicity_of_initials = 1 << (log_n - log_d);
+        coeffs.resize(n, T::zero());
+
+        // swap coefficients in place
+        for i in 0..num_coeffs as u64 {
+            let ri = fft::bitrev(i, log_n);
+            if i < ri {
+                coeffs.swap(i as usize, ri as usize);
+            }
+        }
+
+        // duplicate initial values
+        if duplicity_of_initials > 1 {
+            for i in (0..n).step_by(duplicity_of_initials) {
+                let v = coeffs[i];
+                coeffs[i + 1..i + duplicity_of_initials].fill(v);
```

Comment:
I think this is equivalent? 

It's also parallelizable, so might be a tad bit faster. There are also potentially other parallelizations that could be better (e.g. just chunking `coeffs` up into `num_cpu` chunks, and allocating each chunk to a thread to process serially.)

```suggestion
            ark_std::cfg_chunks_mut!(coeffs, duplicity_of_initials).for_each(|c| {
            	let v = c[i];
                c[i + 1][.. duplicity_of_initials].fill(v);
            });
```